scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_view.py

```diff
@@ -207,24 +207,32 @@ def set_axis_ylimits(self, axis_number, ylims):
     def set_axes_limits(self, xlim, ylim):
         plt.setp(self.fig.axes, xlim=xlim, ylim=ylim)
 
-    def autoscale_y_axes(self):
+    def autoscale_y_axes(self,yerr=0):
         ymin = 1e9
         ymax = -1e9
+
         for axis in self.fig.axes:
             ymin_i, ymax_i = self._get_y_axis_autoscale_limts(axis)
             if ymin_i < ymin:
                 ymin = ymin_i
             if ymax_i > ymax:
                 ymax = ymax_i
 
+        ymax,ymin = ymax + yerr , ymin - yerr
```

Comment:
I think here we should have
```suggestion
        ymax,ymin = 1.1*(ymax + yerr) , 0.9*(ymin - yerr)
```
That way we have some space from the end of error bars and the edge of the plot

Reply:
That is true. If we do it in a few steps:

shift = abs(ymax+yerr)*1.1

then we just add/subtract the shift as appropriate.

Reply:
we should be looking the max `y[i]+y_err[i]` etc. and not the largest y value plus some error. The errors are different for each data point. So the largest sum may have a small y value if the error is huge.